control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2083,7 +2083,10 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 }
 
 func (r *HostedControlPlaneReconciler) reconcileManagedEtcd(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImageProvider *imageprovider.ReleaseImageProvider, createOrUpdate upsert.CreateOrUpdateFN, statefulSet *appsv1.StatefulSet) error {
-	p := etcd.NewEtcdParams(hcp, releaseImageProvider)
+	p, err := etcd.NewEtcdParams(hcp, releaseImageProvider)
+	if err != nil {
+		return fmt.Errorf("error creating etcd params")
```

Comment:
Should you pass the error along here?

Reply:
Yep, I think we need to bubble the error and avoid to continue with the HCP Deployment progress. It should not be malformed, because the API machinery will take care of that on the unmarshal, but I prefer to fail soon than later and let the user to reconfigure the HC manifest